plugin/trace/trace.go

```diff
@@ -81,34 +89,29 @@ func (t *trace) Name() string { return "trace" }
 
 // ServeDNS implements the plugin.Handle interface.
 func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
-	trace := false
-	if t.every > 0 {
-		queryNr := atomic.AddUint64(&t.count, 1)
-
-		if queryNr%t.every == 0 {
-			trace = true
-		}
-	}
 	span := ot.SpanFromContext(ctx)
-	if !trace || span != nil {
+	if span != nil {
 		return plugin.NextOrFailure(t.Name(), t.Next, ctx, w, r)
 	}
 
 	req := request.Request{W: w, Req: r}
-	span = t.Tracer().StartSpan(spanName(ctx, req))
+	span = t.Tracer().StartSpan("servedns")
 	defer span.Finish()
 
 	rw := dnstest.NewRecorder(w)
 	ctx = ot.ContextWithSpan(ctx, span)
 	status, err := plugin.NextOrFailure(t.Name(), t.Next, ctx, rw, r)
 
+	span.SetTag(tagClient, req.IP())
 	span.SetTag(tagName, req.Name())
-	span.SetTag(tagType, req.Type())
+	span.SetTag(tagProto, req.Proto())
 	span.SetTag(tagRcode, rcode.ToString(rw.Rcode))
+	span.SetTag(tagType, req.Type())
+	if t.EndpointType == "datadog" {
+		// We sample client side so we want to analyse every trace we send.
+		// This will create usefull metrics for graphs
+		span.SetTag(ext.AnalyticsEvent, true)
```

Comment:
We shouldn't do it by default. It'd means that 100% of these spans will reach the backend and be indexed (which is a paying feature).

This is usually something configurable. https://docs.datadoghq.com/tracing/app_analytics/. For instance, looking at the `DD_TRACE_ANALYTICS_ENABLED` env var.

Reply:
Using an env var for configuring coredns is a bit weird for something that has a huge config file surface.

Changing the module configuration to include a datadog specific interface might be something to study.
I would defer to upstream to have an answer that match what their feeling is on the subject.

Reply:
We probably want to make it configurable if it is a paying feature right?
So either:
- add an option to the trace plugin (the "standard" coredns way)
- add an env variable (the "standard" datadog way)

I'm inclined to use the first one which will be easier to get accepted upstream. Wdyt?

Reply:
The first one is the one i'm leaning toward too.
AFAIK the only other plugin that are able to use env var are the cloud provider ones
Because the libraries preempt the config in a way.

Reply:
Going with a native option is certainly the way to go.

But before considering using "analytics enabled", could we instead configure the regular sampling rate (`t.every` value?).

Reply:
That's already the case , every trace is sent to be analyzed , but we only send a trace for `t.every` request.